pandas/io/stata.py

```diff
@@ -626,10 +626,10 @@ def _cast_to_stata_types(data: DataFrame) -> DataFrame:
                     ws = precision_loss_doc.format("int64", "float64")
         elif dtype in (np.float32, np.float64):
             value = data[col].max()
```

Comment:
Nit: Could you move this line after the `raise ValueError` block since it's not needed in the check?

Reply:
No problem.